control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -376,6 +379,96 @@ impl ResourcePublishing for OperationGuardArc<VolumeSpec> {
 
         self.complete_update(registry, result, spec_clone).await
     }
+
+    async fn republish(
+        &mut self,
+        registry: &Registry,
+        request: &Self::Republish,
+    ) -> Result<Self::PublishOutput, SvcError> {
+        let specs = registry.specs();
+        let state = registry.get_volume_state(&request.uuid).await?;
+
+        // If the volume is not published then it should issue publish call rather than republish.
+        if state.target.is_none() {
+            return Err(SvcError::VolumeNotPublished {
+                vol_id: state.uuid.to_string(),
+            });
+        }
+
+        let nexus_node = get_volume_target_node(registry, &state, request, true).await?;
+        let newer_nexus_id = NexusId::new();
+
+        let operation =
+            VolumeOperation::Republish((nexus_node.clone(), newer_nexus_id.clone(), request.share));
+        let spec_clone = self.start_update(registry, &state, operation).await?;
+
+        let volume_target = spec_clone.target.as_ref().expect("already validated");
+        let result = specs.nexus(volume_target.nexus()).await;
+        let mut older_nexus = self
+            .validate_update_step(registry, result, &spec_clone)
+            .await?;
+
+        let older_nexus_id = older_nexus.uuid().clone();
+        // shutdown the older nexus before newer nexus creation.
+        let result = older_nexus
+            .shutdown(registry, &ShutdownNexus::new(older_nexus_id))
+            .await;
+        self.validate_update_step(registry, result, &spec_clone)
+            .await?;
```

Comment:
What happens if the node is offline and we can't shutdown? How should we handle this?
We can leave it for another PR? wdyt?

Reply:
Yes, right, this will error out if node goes offline, we need some way to handle this. Maybe we should allow grpc server not implemented errors and use the reservation keys to handle that scenario, and for any other error we error out?

Reply:
For now have added a comment. Needs revisiting again.